internal/provider/vapp/common_org_network.go

```diff
@@ -59,27 +42,27 @@ func (s *orgNetworkResourceModel) initOrgNetworkQuery(ctx context.Context, clien
 		}
 	}
 
-	orgNetworkRef := &orgNetworkRef{
+	networkRef := &networkRef{
 		VDC: vdc,
 	}
 
 	if lockVApp {
-		orgNetworkRef.VAppRef = vapp.Ref{
+		networkRef.VAppRef = vapp.Ref{
 			Name:  s.VAppName.ValueString(),
 			Org:   client.GetOrg(),
 			VDC:   s.VDC.ValueString(),
 			TFCtx: ctx,
 		}
 
-		if errLock := orgNetworkRef.VAppRef.LockParentVApp(); errors.Is(errLock, vapp.ErrVAppRefEmpty) {
+		if errLock := networkRef.VAppRef.LockParentVApp(); errors.Is(errLock, vapp.ErrVAppRefEmpty) {
 			return nil, &diagnosticError{
 				Summary: "Error locking vApp",
 				Detail:  "Empty name, org or vdc in vapp.VAppRef",
 			}
 		}
-		orgNetworkRef.VAppLocked = true
-		orgNetworkRef.VAppUnlockF = func() {
-			if errUnlock := orgNetworkRef.VAppRef.UnLockParentVApp(); errUnlock != nil {
+		networkRef.VAppLocked = true
+		networkRef.VAppUnlockF = func() {
+			if errUnlock := networkRef.VAppRef.UnLockParentVApp(); errUnlock != nil {
 				// tflog print error is enough ?
 				tflog.Warn(ctx, "Error unlock vapp")
 			}
```

Comment:
Really need to unlock the vApp if this one is locked ? It's an error if the vApp is already locked, no ?

Reply:
Is this the way that VCD provider work to.